docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,119 @@
+
+#### additionalClusterDeploymentLabels
+Deprecate current implementation of hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].additionalClusterDeploymentLabels. It's implementation will not change.
+
+#### clusterDeploymentLabelSelector
+This would be a new feature, of type [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), and we'd use the LabelSelector.MatchLabels and/or LabelSelector.MatchExpressions to match the conditions in order to decide if a metric should be reported.
+This encapsulates slightly more advanced filter logic over the existing clusterdeployment labels. In the example above, hive_foo_counter metric will only be reported for the cluster deployments that are labelled with aro-snowflake, not in limited support, and has the metric value > 10m.
+
+### Implementation Details / Notes
+- In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.`
+- Current configuration of using a shorthand camelCase key to refer to a corresponding metric while configuring hiveconfig.spec.metricsConfig.metricsWithDuration will be removed, and users will have to use the full metric name in the proposed configuration.
+- Additionally, we would also want to remove `cluster_type` as a fixed label for all the metrics that do report it now. This would be a breaking change as the consumers who want `cluster_type` will have to configure it via metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels for all the metrics they need it for.
+- There are other fixed labels (like platform and version), which are directed fed from the clusterdeployment labels. Ideally we would want to feed them via hiveconfig, but we might need an exception for the cluster deployment labels that are applied by hive. 
+- All the options configured for a metric within clusterDeploymentRelatedMetrics, will work in tandem with each other. For ex, if all possible options are specified for a metric, then that metric will be reported with the additional labels as per additionalClusterDeploymentLabels, and will be reported only if it matches the labels and/or expressions as per clusterDeploymentLabelSelector and if the duration to be reported exceeds the minimumDuration
+- Failure modes:
+  These situations will cause a panic().
+  - clusterDeploymentRelatedMetrics[].name doesn't exist as a metric
+  - clusterDeploymentRelatedMetrics[].name is a metric without duration but minimumDuration was specified
+  - additionalClusterDeploymentLabels conflict with existing fixed label for that metric
+
+### Risks and Mitigations
```

Comment:
Mike (and Toni) reviewed this recently with me, and they raised some concerns about the deprecation of existing implementations and how it would work. It should be easy to detect if `metricsConfig.clusterDeploymentRelatedMetrics` is present or not - and that can dictate the "old" path vs "new" path. Would that be enough? Or should we decide and document more info about it?

Reply:
Oh, look, great minds and all that. TL;DR: Yes. See above.